LeaderboardBackend/Models/Validation/SlugRule.cs

```diff
@@ -0,0 +1,16 @@
+using FluentValidation;
+
+namespace LeaderboardBackend.Models.Validation;
+
+public static class SlugRule
+{
+    public const string SLUG_FORMAT = "SlugFormat";
+
+    public static IRuleBuilderOptions<T, string> Slug<T>(this IRuleBuilder<T, string> ruleBuilder)
+        => ruleBuilder
+            .Length(2, 80)
+                .WithErrorCode(SLUG_FORMAT)
+            .Matches(@"^([a-zA-Z0-9-_]|%[A-F0-9]{2})*$")
+                .WithErrorCode(SLUG_FORMAT)
+                .WithMessage("Invalid slug format.");
```

Comment:
I brought this up before and basically we have a few options.
|  | Separate Entities/Models | Combined Entities/Models |
| --- | :---: | --- |
| **Validate data as we get it** | | |
| **Validate data in the db**| | |
| **Validate on both ends** | (our setup) | |

Validating data before it even gets to the database is something that I want to keep doing since it improves performance. If you can terminate the request before it even makes a connection to the database then response times will improve. On the other hand, you definitely want to validate on the database side as well so that garbage data that maybe didn't come in through the backend doesn't get persisted. So the only thing that would help then is to merge our DB entities and our DTOs together and simply use annotations and/or fluent validation to control them. I am not opposed to this idea but I think it'll have to be a separate PR.

Reply:
I'll have to see examples of merges, cuz the way I imagine things, you will have fields in DTOs that won't exist in entities and vice versa, e.g. response DTOs will have fields taken from relations. But okay; thanks for the reminder on why we validate this way.